test/cmd_line/read.test.ts

```diff
@@ -0,0 +1,24 @@
+"use strict";
+
+import { ModeHandler } from '../../src/mode/modeHandler';
+import { setupWorkspace, cleanUpWorkspace, assertEqualLines } from './../testUtils';
+import { runCmdLine } from '../../src/cmd_line/main';
+
+suite("read", () => {
+  let modeHandler: ModeHandler;
+
+  setup(async () => {
+    await setupWorkspace();
+    modeHandler = new ModeHandler();
+  });
+
+  teardown(cleanUpWorkspace);
+
+  test("Can read shell command output", async () => {
+    await runCmdLine('r! echo heyho', modeHandler);
```

Comment:
1. This works for me on Ubuntu, probably fails on Windows though :)
2. The error that is thrown is caught in `runCmdLine()` that attempts to display an error message the status bar, this is the same mechanism that would handle  `:thisexcommanddoesnotexist`. I don't see anything in the status bar though, I guess that's another issue.